src/FakeItEasy/Configuration/RuleBuilder.cs

```diff
@@ -222,22 +229,17 @@ public IReturnValueConfiguration<TMember> WhenArgumentsMatch(Func<ArgumentCollec
                 return this;
             }
 
-            public UnorderedCallAssertion MustHaveHappened(int numberOfTimes, Times timesOption)
-            {
-                Guard.AgainstNull(timesOption, nameof(timesOption));
+            public UnorderedCallAssertion MustHaveHappened(int numberOfTimes, Times timesOption) =>
+                this.ParentConfiguration.MustHaveHappened(numberOfTimes, timesOption);

```

Comment:
I thought it reasonable, when we're just deferring to another public method with the same parameters, to let the second method do the null check. But if you worry that we might then lose the check if ever we refactor, I am happy to replace. And to add checks where we don't have them.

Reply:
👍